### DISCUSSION TITLE
"ERR: path has changed" from Android VLC

### DISCUSSION THREAD

**Discussion by JC-85** [bluenviron/mediamtx#265] [Questions]
### Which version are you using?
v0.9.15

### Which operating system are you using?
Windows

### Describe the problem
When connecting to the port with VLC (3.2.12) on Android  I receive the following error message, and the connection fails:

```
2020/10/04 00:28:16 [2/0/0] [client [::1]:52111] connected
2020/10/04 00:28:16 [2/0/0] [client [::1]:52111] OPTIONS
2020/10/04 00:28:16 [2/0/0] [client [::1]:52111] ANNOUNCE
[udp @ 000001d238191640] 'circular_buffer_size' option was set but it is not supported on this build (pthread support is required)
[udp @ 000001d238191e40] 'circular_buffer_size' option was set but it is not supported on this build (pthread support is required)
2020/10/04 00:28:16 [2/0/0] [client [::1]:52111] SETUP
2020/10/04 00:28:16 [2/0/0] [client [::1]:52111] RECORD
2020/10/04 00:28:16 [2/0/0] [client [::1]:52111] is publishing on path 'ch0', 1 track via udp
Output #0, rtsp, to 'rtsp://localhost:8554/ch0':
  Metadata:
    encoder         : Lavf58.29.100
    Stream #0:0: Video: mpeg4, yuv420p, 1280x720, q=2-31, 200 kb/s, 30 fps, 90k tbn, 30 tbc
    Metadata:
      encoder         : Lavc58.54.100 mpeg4
    Side data:
      cpb: bitrate max/min/avg: 0/0/200000 buffer size: 0 vbv_delay: -1
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39856] SETUP
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39856] PLAY
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39856] is receiving on path 'ch0', 1 track via udp
2020/10/04 00:28:16 [2/1/1] [client 192.168.2.34:39856] TEARDOWN
2020/10/04 00:28:16 [1/1/0] [client 192.168.2.34:39856] disconnected
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39858] connected
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39858] SETUP
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39858] PLAY
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39858] ERR: path has changed, was 'ch0', now is 'ch0/trackID=0'
2020/10/04 00:28:16 [1/1/0] [client 192.168.2.34:39858] disconnected
```

Connecting to rtsp-simple-server from VLC (3.0.11) on Windows works without any problems. Connecting to other RTSP streams using the same Android device also works without issues.

**Comment by aler9** [comment]
Hello, i've tested this exact setup (rtsp-simple-server on Windows, VLC on Android on different machine) but everything worked and i was not able to replicate the issue.

In my opinion, there's a firewall between the machine and the device that blocks UDP connections; if there's a firewall, VLC sends a TEARDOWN request after 10 seconds, and then try with TCP, but in this case the teardown happens instanteously:
```
2020/10/04 00:28:16 [2/1/0] [client 192.168.2.34:39856] is receiving on path 'ch0', 1 track via udp
2020/10/04 00:28:16 [2/1/1] [client 192.168.2.34:39856] TEARDOWN
```

There should be a delay of 10 seconds between these two lines.
Please try again and attach another log.

**Comment by VashStan** [comment]
hi,i met this error in all the mediamtx version, and  my device is synology ,i install the mediamtx in the docker, and my synology has ipv6 addr but no ipv4 addr, and i can push rtsp stream to the server(use synology),but  when i use vlc to get rtsp stream(use windows vlc with the internet) , the error "path has changed" come up . i tried to change the udp to tcp , but it's no use,and the error  meeeage become something else(i cant find the log ) . i guess  the problem is because the udp or tcp can not reache the server reightly. 

and .. finally i gave up ,the rtsp and docker is new to me , there are too many things that i dont know ,so... maybe sometime ,i would try it again.

And my English is not  good ，so maybe my description is not so right ,sorry

**Reply by aler9** [reply]
>  i guess the problem is because the udp or tcp can not reache the server reightly.

Disable UDP and force TCP by using the `MTX_PROTOCOLS` environment variable:

```
docker run --rm -it \
-e MTX_PROTOCOLS=tcp \
-p 8554:8554 \
-p 1935:1935 \
-p 8888:8888 \
-p 8889:8889 \
-p 8890:8890/udp \
bluenviron/mediamtx
```

[a comment by an automated account is not shown]
